pkg/patterns/addon/pkg/status/kstatus.go

```diff
@@ -12,7 +12,59 @@ import (
 	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative"
 )
 
+// Returns the set of abnormal conditions for a given resource.
+// abnormal conditions reflect any condition indicating an other than nominal state
+type AbnormalConditionsMethod func(ctx context.Context, unstruct *unstructured.Unstructured) []status.Condition
+
 type kstatusAggregator struct {
+	// Map of GVK specific 'Compute' methods. Methods find the status of
+	// a given unstructured resource.
+	//
+	// The returned result contains the status of the resource, which will be
+	// one of
+	//   - InProgress
+	//   - Current
+	//   - Failed
+	//   - Terminating
+	//
+	// It also contains a message that provides more information on why
+	// the resource has the given status. Finally, the result also contains
+	// a list of standard resources that would belong on the given resource.
+	GVKComputeMethods map[string]status.GetConditionsFn
+
+	// Map of GVK specific methods which return the set of abnormal conditions for a given resource
+	// abnormal conditions reflect any condition indicating an other than nominal state
+	GVKAbnormalConditionsMethods map[string]AbnormalConditionsMethod
+}
+
+// NewOpenKstatusAgregator creates a kstatusAggregator with support for supplied gvk string specific 'Compute' and 'AbnormalConditions' methods
+// Compute and Abnormal conditions methods should be supplied in assocation with a string of an explicit
+// k8s.io/apimachinery/pkg/runtime/schema.GroupVersionKind (GVK) string
+//
+// e.g.
+// computMethods := make(map[string]GetConditionsFn)
+// conditionsMethods := make(map[string]AbnormalConditionsMethod)
```

Comment:
```suggestion
// abnormalConditionsMethods := make(map[string]AbnormalConditionsMethod)
```
Is my suggestion correct?

Reply:
@atoato88 nice catch, you are correct though there were two instances of `conditionsMethods` which need to be updated to `abnormalConditionsMethods` which are now patched as of commit [6de4512](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/372/commits/6de4512dee82ca425f2f2998eb5a76c27dd64cab)